src/client/server_mutate_ticks.rs

```diff
@@ -0,0 +1,335 @@
+    pub fn last_tick(&self) -> RepliconTick {
+        self.last_tick
+    }
+
+    /// Returns a mask that represents the received ticks.
+    pub fn mask(&self) -> u64 {
+        let mut bitmask = 0;
+
+        for (i, tick) in self.ticks.iter().enumerate() {
+            if tick.all_received() {
+                bitmask |= 1 << i;
+            }
+        }
+
+        bitmask
+    }
+
+    /// Returns `true` if this tick is confirmed for an entity.
+    ///
+    /// All ticks older then 64 ticks since [`Self::last_tick`] are considered received.
+    pub fn contains(&self, tick: RepliconTick) -> bool {
+        if tick > self.last_tick {
+            return false;
+        }
+
+        let ago = self.last_tick - tick;
+        if let Some(tick) = self.ticks.get(ago as usize) {
+            tick.all_received()
+        } else {
+            true
+        }
+    }
+
+    /// Returns `true` if any tick in the given range was confirmed for the entity with
+    /// this component.
+    ///
+    /// All ticks older then 64 ticks since [`Self::last_tick`] are considered received.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `debug_assertions` are enabled and
+    /// `start_tick` is greater then `end_tick`.
+    pub fn contains_any(&self, start_tick: RepliconTick, end_tick: RepliconTick) -> bool {
+        debug_assert!(start_tick <= end_tick);
+
+        if start_tick > self.last_tick {
+            return false;
+        }
+        if start_tick <= self.last_tick - self.ticks.len() as u32 {
+            return true;
+        }
+
+        let end_tick = if end_tick < self.last_tick {
+            end_tick
+        } else {
+            self.last_tick
+        };
+
+        // Start and end are reversed because ticks in the
+        // array are stored in decreasing order.
+        let end = (self.last_tick - start_tick) as usize;
+        let start = (self.last_tick - end_tick) as usize;
+        self.ticks[start..=end]
+            .iter()
+            .any(|tick| tick.all_received())
+    }
+
+    /// Confirms a received message a tick and initializes the number of sent messages for it.
+    ///
+    /// Return `true` if the number of received messages matches `messages_count`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `debug_assertions` are enabled and `messages_count` is different
+    /// from the last call or if the number of calls for the same tick exceeds `messages_count`.
+    pub fn confirm(&mut self, tick: RepliconTick, messages_count: usize) -> bool {
+        let len = self.ticks.len();
+        if tick > self.last_tick {
+            let ago = (tick - self.last_tick) as usize;
+            if ago >= len {
+                // If the difference exceeds the size, clear all ticks.
+                self.ticks.fill(Default::default());
+            } else {
+                // Shift all ticks.
+                self.ticks.copy_within(..len - ago, ago);
+                self.ticks[..ago].fill(Default::default());
```

Comment:
What about using a VecDeque?

Reply:
Do you think it will be faster or just more readable?
With array I like that I have fixed size.

Reply:
> Do you think it will be faster or just more readable?

You avoid copying the array. Probably quite small perf effect but might be worth it.